fearless/Modules/Staking/SelectValidatorsFlow/SelectValidatorsConfirm/SelectValidatorsConfirmPresenter.swift

```diff
@@ -156,29 +156,47 @@ extension SelectValidatorsConfirmPresenter: SelectValidatorsConfirmPresenterProt
     }
 
     func proceed() {
+        guard let state = state else {
+            return
+        }
+
         let locale = view?.localizationManager?.selectedLocale ?? Locale.current
 
-        DataValidationRunner(validators: [
+        let spendingAmount: Decimal = !state.hasExistingBond ? state.amount : 0.0
+
+        var validators: [DataValidating] = [
             dataValidatingFactory.has(fee: fee, locale: locale) { [weak self] in
                 self?.interactor.estimateFee()
             },
-            dataValidatingFactory.canNominate(
-                amount: state?.amount,
-                minimalBalance: minimalBalance,
-                minNominatorBond: minNominatorBond,
+
+            dataValidatingFactory.canPayFeeAndAmount(
+                balance: balance,
+                fee: fee,
+                spendingAmount: spendingAmount,
                 locale: locale
             ),
-            dataValidatingFactory.maxNominatorsCountNotReached(
+
+            dataValidatingFactory.maxNominatorsCountNotApplied(
                 counterForNominators: counterForNominators,
                 maxNominatorsCount: maxNominatorsCount,
+                hasExistingNomination: state.hasExistingNomination,
                 locale: locale
             )
-        ]).runValidation { [weak self] in
-            guard let fee = self?.fee else {
-                return
-            }
+        ]
+
+        if !state.hasExistingBond {
```

Comment:
If there is no bond how we can validate that?

Reply:
If we don't have existing bonding then we are going to submit new one with validators and we need to apply validation for the bonding value